packages/client/hmi-client/src/components/models/tera-stratified-model-configuration.vue

```diff
@@ -162,12 +162,24 @@ const configurations = computed<Model[]>(
 	() => modelConfigurations.value?.map((m) => m.configuration) ?? []
 );
 
-const baseModelStates = computed<any>(() =>
-	props.model?.semantics?.span?.[0].system.model.states.map(({ id }) => id)
-);
-const baseModelTransitions = computed<any>(() =>
-	props.model?.semantics?.span?.[0].system.model.transitions.map(({ id }) => id)
-);
+const MAX_DEPTH = 10;
+const baseModel = computed<any>(() => {
+	let level = props.model?.semantics?.span?.[0].system;
+	let c = 0;
+
+	while (c < MAX_DEPTH) {
+		c++;
+		if (level.semantics.span) {
+			level = level.semantics.span[0].system;
+		} else {
+			return level.model;
+		}
+	}
+	return level.model;
+});
```

Comment:
I think we should move this into petrinet-service as `getBaseAMR` or something - mostly so the recursion stuff happens in one place.